src/event.rs

```diff
@@ -222,51 +222,86 @@ pub enum WindowEvent {
-    ///
-    /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
-    /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
-    /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
-    CursorMoved {
+    PointerMoved {
         device_id: DeviceId,
 
         /// (x,y) coords in pixels relative to the top-left corner of the window. Because the range
         /// of this data is limited by the display area and it may have been transformed by
         /// the OS to implement effects such as cursor acceleration, it should not be used
         /// to implement non-cursor-like interactions such as 3D camera control.
+        ///
+        /// ## Platform-specific
+        ///
+        /// **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
+        ///
+        /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
+        /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
+        /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
         position: PhysicalPosition<f64>,
+
+        source: PointerSource,
     },
 
     /// The cursor has entered the window.
-    ///
-    /// ## Platform-specific
-    ///
-    /// - **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
-    ///
-    /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
-    /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
-    /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
-    CursorEntered { device_id: DeviceId },
+    PointerEntered {
+        device_id: DeviceId,
+
+        /// The position of the cursor when it entered the window.
+        ///
+        /// ## Platform-specific
+        ///
+        /// **Orbital: Always emits `(0., 0.)`.
+        /// **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
+        ///
+        /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
+        /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
+        /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
+        position: PhysicalPosition<f64>,
+
+        ty: PointerType,
+    },
 
     /// The cursor has left the window.
-    ///
-    /// ## Platform-specific
-    ///
-    /// - **Web:** Doesn't take into account CSS [`border`], [`padding`], or [`transform`].
-    ///
-    /// [`border`]: https://developer.mozilla.org/en-US/docs/Web/CSS/border
-    /// [`padding`]: https://developer.mozilla.org/en-US/docs/Web/CSS/padding
-    /// [`transform`]: https://developer.mozilla.org/en-US/docs/Web/CSS/transform
-    CursorLeft { device_id: DeviceId },
+    PointerLeft {
+        device_id: DeviceId,
+
+        /// The position of the cursor when it left the window.
```

Comment:
This definition is troublesome: If the cursor has left the window, then it will be outside the window's bounds, and the position might be non-sensical for the current application.

I see a few options (in order of my preference):
- Define that pointer positions in general can be negative and outside the bounds of the window (on some platforms).
- "Clamp" the position to be within the bounds of the window.
- Define this event to be "the position the cursor had just before it left the window", i.e. store pointer events, and re-emit the position from the last `PointerMoved`.

Reply:
Will report back after some testing how other backends behave exactly, but would go with option 1.

Reply:
Apparently all backends report a position outside the window, which does make sense actually.

Fixed MacOS as well!